src/routes/auth.routes.ts

```diff
@@ -7,25 +7,28 @@ import {
   verify2FAToken,
   loginUser,
   logoutUser,
-} from '../controllers';
-import { ValidateJoi } from '../middleware/validation/validation.middleware';
-import { UserSchema } from '../middleware/validation/user.schema.middleware';
-import { isAuth } from '../middleware/auth';
+} from "../controllers";
+import { ValidateJoi } from "../middleware/validation/validation.middleware";
+import { UserSchema } from "../middleware/validation/user.schema.middleware";
+import { isAuth } from "../middleware/auth";
+import { updatePassword } from "../controllers/updatePassword.controller";
 
 const authRouter = Router();
 
-authRouter.post('/login', ValidateJoi(UserSchema.loginData.create), loginUser); // login a User
+authRouter.post("/login", ValidateJoi(UserSchema.loginData.create), loginUser); // login a User
 
-authRouter.get('/logout', [isAuth], logoutUser); //logout a user
+authRouter.get("/logout", [isAuth], logoutUser); //logout a user
 
-authRouter.get('/confirm/:confirmationCode', confirmUser); // Confirm the user who registered
+authRouter.get("/confirm/:confirmationCode", confirmUser); // Confirm the user who registered
 
-authRouter.post('/auth/requestResetPassword', resetPasswordRequestController); // Request a reset password
+authRouter.post("/auth/requestResetPassword", resetPasswordRequestController); // Request a reset password
 
-authRouter.post('/auth/resetPassword/:token', resetPasswordController); // Reset the password
+authRouter.post("/auth/resetPassword/:token", resetPasswordController); // Reset the password
 
-authRouter.post('/auth/2fa', create2FAToken); // Create a 2FA token
+authRouter.post("/auth/2fa", create2FAToken); // Create a 2FA token
 
-authRouter.post('/auth/2fa/verify2FAToken', verify2FAToken); // Verify the 2FA token
+authRouter.post("/auth/2fa/verify2FAToken", verify2FAToken); // Verify the 2FA token
+
+authRouter.post("/updatepassword", isAuth, updatePassword);
```

Comment:
You forgot to use the middleware here on the route. 😄 
There is currently only the isAuth, try adding `ValidateJoi(UserSchema.passwordUpdate`... 🔒 🔑

Reply:
that is not even working it is in other task don't worry about it